### ISSUE TITLE
[list]: Screen reader announcement issue for additionalText and lack of announcement for selected record

### ISSUE THREAD

**Issue by wilson-xu-sap** [SAP/ui5-webcomponents-react#6988]
### Describe the bug

1. There is no way to set accessible name for additionalText now. `additionalText` should not join with text directly sometimes.
2. There is no way to set detailed announcement after selecting one item. Only `selected` is announced.

<img width="1035" alt="Image" src="https://github.com/user-attachments/assets/a3b57582-dc29-41bd-8a62-45d760a10eaa" />

### Isolated Example

https://sap.github.io/ui5-webcomponents-react/v2/?path=/docs/data-display-list--docs

### Reproduction steps

1. Click on one of the list item
2. use arrow keys to navigate
3. check the screen reader
...

### Expected Behaviour

1. A way to set accessible name for additionalText.
2. Screen reader users should receive an announcement indicating which record is currently selected when navigating the list.

### UI5 Web Components for React Version

2.7

### UI5 Web Components Version

2.7

### Browser

Edge

### Relevant log output

```shell

```

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by Lukas742** [comment]
Thanks for reporting! I'll forward this issue to our UI5 Web Components Colleagues as the affected component is developed in their repository.
